src/main/java/com/fasterxml/jackson/core/io/CharBufferReader.java

```diff
@@ -0,0 +1,66 @@
+package com.fasterxml.jackson.core.io;
+
+import java.io.Reader;
+import java.nio.CharBuffer;
+
+public class CharBufferReader extends Reader {
```

Comment:
Don't do anything yet but since CharBufferReader is only used in one place - I think it would be a good idea to move it the same package as ByteSourceJsonBootstrapper and to make it package private and final to discourage its use by other users. We should at least add javadoc to the class saying that this class is only for internal use of jackson-core.

Reply:
Will do, wanted to figure out where all this might be needed then encapsulate.